internal/elasticsearch/ingest/datastream.go

```diff
@@ -192,20 +192,27 @@ func loadRoutingRuleFile(dataStreamPath string) ([]map[string]interface{}, error
 			}
 
 			processor := make(map[string]interface{})
-			processor["reroute"] = RerouteProcessor{
+			rp := RerouteProcessor{
 				Tag:       r.SourceDataset,
-				If:        rule.If,
 				Dataset:   td,
 				Namespace: ns,
 			}
+			if rule.If != "" {
+				rp.If = rule.If
+			}
+			processor["reroute"] = rp
 			rerouteProcessors = append(rerouteProcessors, processor)
 		}
 	}
 	return rerouteProcessors, nil
 }
 
-func convertValue(value interface{}, label string) ([]string, error) {
-	switch value := value.(type) {
+func convertValue(originalValue interface{}, label string) ([]string, error) {
+	if originalValue == nil {
+		return nil, nil
+	}
+
+	switch value := originalValue.(type) {
```

Comment:
Nit. Make this a case in the switch?
```suggestion
	switch value := originalValue.(type) {
	case nil:
		return nil, nil
```

Reply:
Oh yes what am I thinking?! Thanks!!